### PR TITLE
Split internal settings into settings_internal.js

### PR DESCRIPTION
**Pull request by sbc100**
This allows to add and remove internal settings without effecting the
public ABI and potentially breaking users.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -411,6 +411,9 @@ def standardize_setting_change(key, value):
     key, value = change.split('=', 1)
     key, value = standardize_setting_change(key, value)
 
+    if key in shared.Settings.internal_settings:
+      exit_with_error('%s is an internal setting.  Its should not be set from the command line.', key)
```
Comment (kripken) [inline emcc.py]: `Its` => `It`.

File: src/settings.js [hunk 2]
```diff
@@ -1520,32 +1523,6 @@ var WASM_TABLE_SIZE = 0;
...
-var OFFSCREEN_FRAMEBUFFER_FORBID_VAO_PATH = 0;
-
-// Internal (testing only): Forces memory growing to fail.
-var TEST_MEMORY_GROWTH_FAILS = 0;
-
-// Advanced: Customize this array to reduce the set of asm.js runtime variables
-// that are generated. This allows minifying extra bit of asm.js code from unused
-// runtime code, if you know some of these are not needed.
-// (think of this as advanced manual DCE)
-var ASM_PRIMITIVE_VARS = ['__THREW__', 'threwValue', 'setjmpId', 'tempInt', 'tempBigInt', 'tempBigIntS', 'tempValue', 'tempDouble', 'tempFloat', 'tempDoublePtr', 'STACKTOP', 'STACK_MAX']
```
Comment (kripken) [inline src/settings.js]: This isn't an internal API, @juj uses it I believe.

File: src/settings.js [hunk 3]
```diff
@@ -1559,101 +1536,11 @@ var MINIMAL_RUNTIME = 0;
 // ignored when building with -s MINIMAL_RUNTIME=0.
 var USES_DYNAMIC_ALLOC = 1;
 
-// Advanced manual dead code elimination:
-// Specifies the set of runtime JS functions that should be imported to the asm.js/wasm module.
-// Remove elements from this list to make build smaller if some of these are not needed.
-// In Wasm -O3/-Os builds, adjusting this is not necessary, as the Meta-DCE pass is able to
-// remove these, but if you are targeting asm.js or doing a -O2 build or lower, then this can
-// be beneficial.
-var RUNTIME_FUNCS_TO_IMPORT = ['abort', 'setTempRet0', 'getTempRet0']
```
Comment (kripken) [inline src/settings.js]: Also not internal.

File: src/settings.js [hunk 4]
```diff
@@ -1559,101 +1536,11 @@ var MINIMAL_RUNTIME = 0;
...
-
-// Internal: stores the base name of the output file (-o TARGET_BASENAME.js)
-var TARGET_BASENAME = '';
-
-// If true, compiler supports setjmp() and longjmp(). If false, these APIs are not available.
-// If you are using C++ exceptions, but do not need setjmp()+longjmp() API, then you can set
-// this to 0 to save a little bit of code size and performance when catching exceptions.
-var SUPPORT_LONGJMP = 1;
```
Comment (kripken) [inline src/settings.js]: Also not internal.

File: src/settings.js [hunk 5]
```diff
@@ -1559,101 +1536,11 @@ var MINIMAL_RUNTIME = 0;
...
 // there is no "Module.canvas" object, no magic "null" default handling, and DOM element
 // 'target' parameters are taken to refer to CSS selectors, instead of referring to DOM IDs.
 var DISABLE_DEPRECATED_FIND_EVENT_TARGET_BEHAVIOR = 0;
 
-// Specifies whether the generated .html file is run through html-minifier. The set of
-// optimization passes run by html-minifier depends on debug and optimization levels. In
-// -g2 and higher, no minification is performed. In -g1, minification is done, but whitespace
-// is retained. Minification requires at least -O1 or -Os to be used. Pass -s MINIFY_HTML=0
-// to explicitly choose to disable HTML minification altogether.
-var MINIFY_HTML = 1;
```
Comment (kripken) [inline src/settings.js]: Also not internal.

File: src/settings.js [hunk 6]
```diff
@@ -1559,101 +1536,11 @@ var MINIMAL_RUNTIME = 0;
...
-// wasm backend is in use with WASM=0 (to enable non-wasm output, we compile to
-// wasm normally, then compile that to JS).
-var WASM2JS = 0;
-
-// Whether we *may* be using wasm2js. This compiles to wasm normally, but lets you
-// run wasm2js *later* on the wasm, and you can pick between running the normal
-// wasm or that wasm2js code. For details of how to do that, see the test_maybe_wasm2js
-// test.
-// This option can be useful for debugging and bisecting.
-var MAYBE_WASM2JS = 0;
```
Comment (kripken) [inline src/settings.js]: Also not internal.

File: src/settings.js [hunk 7]
```diff
@@ -1559,101 +1536,11 @@ var MINIMAL_RUNTIME = 0;
...
-// by -fsanitize=leak instead of used directly.
-var USE_LSAN = 0;
-
-// Whether we should link in ASan's runtime library. This is intended to be used
-// by -fsanitize=leak instead of used directly.
-var USE_ASAN = 0;
-
-// The size of our shadow memory.
-// By default, we have 32 MiB. This supports 256 MiB of real memory.
-var ASAN_SHADOW_SIZE = 33554432;
```
Comment (kripken) [inline src/settings.js]: I think this is not an internal API, but am not sure.

File: src/settings.js [hunk 8]
```diff
@@ -1559,101 +1536,11 @@ var MINIMAL_RUNTIME = 0;
...
-
-// Whether we should load the WASM source map at runtime.
-// This is enabled automatically when using -g4 with sanitizers.
-var LOAD_SOURCE_MAP = 0;
-
-// Whether we should use the offset converter.
-// This is needed for older versions of v8 (<7.7) that does not give the hex module offset
-// into wasm binary in stack traces, as well as for avoiding using source map
-// entries across function boundaries.
-var USE_OFFSET_CONVERTER = 0;
```
Comment (kripken) [inline src/settings.js]: Also not sure.

**Comment by sbc100** [comment]
Yeah, I basically took everything below the "Things below here are internal" and put them in the internal file.

Clearly that was not accurate and I was going to let the CI give me a first approximation.

**Pull request merged**